NETunnelProvider.HandleAppMessage() method signature does not match underlying Obj-C Implementation

<!-- 🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨 -->

<!--
Please fill in the *entire* template below.
The template is designed to avoid unnecessary delays to confirm and fix reported issues. Issues with missing information generally end up with a `need-info` label and take significantly more time to fix. Please help us help you!
For more information see https://github.com/xamarin/xamarin-macios/wiki/Submitting-Bugs-&-Suggestions
For support requests use https://www.xamarin.com/support
-->
### Steps to Reproduce

1. Implement the method as defined in [networkextension.cs](https://github.com/xamarin/xamarin-macios/blob/main/src/networkextension.cs)
2. Attempt to pass a null value to the completion handler.

### Expected Behavior
Null values should be able to be passed to the completion handler. This is implied in the Obj-C declaration, but is explicit in the swift declaration.

```
func handleAppMessage(
    _ messageData: Data,
    completionHandler: ((Data?) -> Void)? = nil
)
```
### Actual Behavior
If implement as defined in the C# NETunnelProvider interface null values are not allowed to be passed to the competition handler. 
This can be worked around by modifying the overridden method signature (below) to allow null values to be passed, but this results in a compiler warning.

```
public override void HandleAppMessage(NSData messageData, Action<NSData?>? completionHandler)
{
}
```
### Environment

<details>
<summary>Version information</summary>

```
Visual Studio for Mac
17.5 Preview (17.5 build 437)
```

</details>

### Build Logs
```
<redacted>/PacketTunnelProvider.cs(91,26): warning CS8610: Nullability of reference types in type of parameter 'completionHandler' doesn't match overridden member. [<redacted>.csproj::TargetFramework=net7.0-ios]
```

--- a/src/networkextension.cs
+++ b/src/networkextension.cs
@@ -2097,6 +2097,10 @@ namespace NetworkExtension {
 		NEProxySettings ProxySettings { get; set; }
 	}
 
+#if XAMCORE_5_0
+	delegate void NETunnelProviderHandleAppMessageCallback ([NullAllowed] NSData data);
+#endif
+
 	/// <summary>Base class for extensions that implement client-side ends of a network tunnel.</summary>
 	/// <remarks>To be added.</remarks>
 	/// <related type="externalDocumentation" href="https://developer.apple.com/library/ios/documentation/NetworkExtension/Reference/NETunnelProviderClassRef/index.html">Apple documentation for <c>NETunnelProvider</c></related>
@@ -2120,7 +2124,11 @@ namespace NetworkExtension {
 			        </returns>
 			<remarks>To be added.</remarks>
 			""")]
+#if XAMCORE_5_0
+		void HandleAppMessage (NSData messageData, [NullAllowed] NETunnelProviderHandleAppMessageCallback completionHandler);
+#else
 		void HandleAppMessage (NSData messageData, [NullAllowed] Action<NSData> completionHandler);
+#endif
 
 		/// <param name="tunnelNetworkSettings">
 		///           <para>To be added.</para>

